src/shared/components/query/QueryStore.ts

```diff
@@ -143,126 +143,43 @@ export class QueryStore
 	}
 
 	@observable studiesHaveChangedSinceInitialization:boolean = false;
-	@observable savedVirtualCohorts:VirtualCohort[] = [];
-
-	@action public deleteVirtualCohort(id:string) {
-		VirtualCohorts.delete(id);
 
-		this.loadSavedVirtualCohorts();
-	}
+	@observable deletedVirtualStudies:string[] = [];
 
-	@action private loadSavedVirtualCohorts() {
-		let localStorageVirtualCohorts:LocalStorageVirtualCohort[] = VirtualCohorts.get();
-		this.savedVirtualCohorts = localStorageVirtualCohorts.map((x:any)=>{
-			let samples:{studyId:string, sampleId:string}[] = [];
-			const constituentStudyIds:string[] = [];
-			for (const selectedCasesObj of x.selectedCases) {
-				samples = samples.concat(selectedCasesObj.samples.map((sampleId:string)=>({studyId:selectedCasesObj.studyID, sampleId})));
-				constituentStudyIds.push(selectedCasesObj.studyID);
-			}
-			return {
-				id: x.virtualCohortID,
-				name: x.studyName,
-				description: x.description,
-				samples,
-				constituentStudyIds
-			};
-		});
-	}
+	@action

```

Comment:
you should not put @action decorator on an async method.   The actual mutation of state will end up executing outside of the action, thus nullifying purpose of action.   See documentation for further explaination.  https://mobx.js.org/best/actions.html.   i would reccomend using action() around the actual mutation  (inside callback)